src/java.base/share/classes/java/util/Base64.java

```diff
@@ -810,12 +810,12 @@ private int decode0(byte[] src, int sp, int sl, byte[] dst) {
 
             while (sp < sl) {
                 if (shiftto == 18 && sp < sl - 4) {       // fast path
-                    int dl = decodeBlock(src, sp, sl, dst, dp, isURL);
+                    int dl = decodeBlock(src, sp, sl, dst, dp, isURL, isMIME);
                     /*
                      * Calculate how many characters were processed by how many
                      * bytes of data were returned.
                      */
-                    int chars_decoded = (dl / 3) * 4;
+                    int chars_decoded = ((dl + 2) / 3) * 4;
```

Comment:
In the PR comments, you say, "A change was also made here removing the restriction that the intrinsic must return an even multiple of 3 bytes decoded.", however there's still a comment in the code above that says:

         * If the intrinsic function does not process all of the bytes in
         * src, it must process a multiple of four of them, making the
         * returned destination length a multiple of three.
        
So this comment needs to be changed or removed to reflect your commit.

Reply:
I will change the comment, and add verbage regarding the new parameter.  Thank you.